CADETProcess/processModel/flowSheet.py

```diff
@@ -406,9 +524,27 @@ def connection_exists(self, origin, destination):
         destination : UnitBaseClass
             UnitBaseClass where the connection terminates.
 
+        Notes
+        -----
+
+
         """
-        if destination in self._connections[origin].destinations \
-                and origin in self._connections[destination].origins:
+        if origin.has_ports and not origin_port:
+            raise CADETProcessError(f'{origin.name} is a Unit Operation with ports, so you need to specify origin_port.')
```

Comment:
[alternative] Consider updating the error message, e.g. "Origin port needs to be specified for unit operations with ports."

Also see below for destination port.